saml/response.go

```diff
@@ -19,19 +19,21 @@ import (
 )
 
 type parseResponseOptions struct {
-	clock                            clockwork.Clock
-	skipRequestIDValidation          bool
-	skipAssertionConditionValidation bool
-	skipSignatureValidation          bool
-	assertionConsumerServiceURL      string
+	clock                                   clockwork.Clock
+	skipRequestIDValidation                 bool
+	skipAssertionConditionValidation        bool
+	skipSignatureValidation                 bool
+	assertionConsumerServiceURL             string
+	requireSignatureForResponseAndAssertion bool
```

Comment:
For `requireSignatureForResponseAndAssertion` :
On the vault usage side i opted for a name `EnableStrictResponseSignatureValidation` as that sounded more user friendly for a user exposed setting. The description will share the impact of the setting. Whereas kept it `requireSignatureForResponseAndAssertion` on cap side thinking of it as more of an internal lib so can have any name. 

Any preference of using `EnableStrictResponseSignatureValidation`  here?

Reply:
I don't have a strong preference. Based on my [comment](https://github.com/hashicorp/cap/pull/145#discussion_r1867487684) about having all 3 options signature validation, it would just be important to think about the general naming pattern:

1. Validate Response And Assertion Signatures = `EnableStrictSignatureValidation`
2. Validate Response Signature = `EnableResponseSignatureValidation`?

Alternatively, we could use a more explicit naming pattern:

1. `ValidateResponseSignature`
3. `ValidateResponseAndAssertionSignatures`

I'm ok either way.

Reply:
Updated the name of the setting.